### ISSUE TITLE
Abstract hashCode and equals breaks Jackson compatibility

### ISSUE THREAD

**Issue by ocrease** [immutables/immutables#300]
I have a situation where my Abstract Value Type extends an abstract class that has non-default `equals` and `hashCode` implementations. Here is an example:

```java
import org.immutables.value.Value;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;

public class Sample
{
    public static abstract class AbstractBaseClass {
        @Override
        public int hashCode() { return 1; }

        @Override
        public boolean equals(Object other) { return false; }
    }
    
    @Value.Immutable
    @JsonSerialize
    public static abstract class SubClass extends AbstractBaseClass {
        public abstract int getA();
    }
}
```

During compilation I get the following warnings:

```
[ERROR] Sample.java:[28,27] Type inherits non-default 'hashCode' method but have some non-inherited attributes. Please override 'hashCode' with abstract method to have generated it. Otherwise override with calling super implemtation to usecustom implementation
[ERROR] Sample.java:[28,27] Type inherits non-default 'equals' method but have some non-inherited attributes. Please override 'equals' with abstract method to have generate it. Otherwise override with calling super implemtation to use custom implementation
```

If I add the methods as suggested:

```java
@Value.Immutable
@JsonSerialize
public static abstract class SubClass extends AbstractBaseClass {
    public abstract int getA();
    
    public abstract int hashCode();
    public abstract boolean equals(Object other);
}
```

Then I get the following error:
```
[ERROR] ImmutableSubClass.java:[95,15] error: Json is not abstract and does not override abstract method hashCode() in SubClass
```

The generated `Json` class within the `ImmutableSubClass` doesn't compile because it does not override the hashCode method.

```java
static final class Json extends Sample.SubClass {
  @Nullable Integer a;
  @JsonProperty
  public void setA(int a) {
    this.a = a;
  }
  @Override
  public boolean equals(Object other) { throw new UnsupportedOperationException(); }
  @Override
  public int getA() { throw new UnsupportedOperationException(); }
}
```

This means I have to choose between having the equals/hashcode methods or not using Jackson. I could implement the hashCode and equals manually, but that is a bit tiresome!

**Comment by elucash** [comment]
Thank you for reporting this, definitely should be fixed!

**Comment by elucash** [comment]
Released in v 2.1.16 Thank you!
